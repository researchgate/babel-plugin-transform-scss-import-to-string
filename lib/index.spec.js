const path = require('path');
const babel = require('babel-core');
const plugin = require('./');

const transformFixture = (fixture, options = {}) => {
  const fixturePath = path.resolve(__dirname, '__fixtures__', fixture);
  const { code } = babel.transformFileSync(fixturePath, {
    plugins: [[plugin, options]],
  });
  return code;
};

describe('babel-plugin-transform-scss-import-to-string', () => {
  it('resolves and transpiles scss file', () => {
    expect(transformFixture('sample.js')).toMatchSnapshot();
  });

  it('skips non-default import specifier', () => {
    expect(transformFixture('invalid.js')).toMatchSnapshot();
  });

  it('supports custom node-sass options', () => {
    expect(
      transformFixture('precision.js', { precision: 3 })
    ).toMatchSnapshot();
  });

  it('supports nested scss files', () => {
    expect(() => transformFixture('nested.js')).not.toThrow();
  });
});
