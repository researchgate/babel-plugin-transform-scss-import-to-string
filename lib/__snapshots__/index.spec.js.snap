// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-scss-import-to-string resolves and transpiles scss file 1`] = `
"const sideEffectStyles = \\".foo {\\\\n  color: cyan; }\\\\n\\\\n.bar {\\\\n  font-size: 32px; }\\\\n\\";

if (typeof sideEffectStyles === 'string') {
  const style = document.createElement('style');
  style.styleSheet.cssText = sideEffectStyles;
  document.head.appendChild(style);
}"
`;

exports[`babel-plugin-transform-scss-import-to-string skips non-default import specifier 1`] = `
"import { foo } from './sample.scss';
console.log(foo);"
`;

exports[`babel-plugin-transform-scss-import-to-string supports custom node-sass options 1`] = `"const precision = \\".baz {\\\\n  transform: translateX(1.877px); }\\\\n\\";"`;
